lang/csharp/src/apache/main/CodeGen/CodeGen.cs

```diff
@@ -219,8 +219,14 @@ protected virtual void processProtocols()
         /// <returns>
         /// List of named schemas.
         /// </returns>
-        protected virtual SchemaNames generateNames(Protocol protocol)
+        /// <exception cref="System.ArgumentNullException">protocol - Protocol can not be null.</exception>
+        protected virtual SchemaNames GenerateNames(Protocol protocol)
```

Comment:
For backward compatibility we need to keep the old method deprecated for few releases.

Reply:
Fixed